### ISSUE TITLE
xcode build failed when I use 'static_framework' . ld: library not found for 

### ISSUE THREAD

**Issue by heroims** [CocoaPods/CocoaPods#7352]
* [x] I've read and understood the [*CONTRIBUTING guidelines and have done my best effort to follow](https://github.com/CocoaPods/CocoaPods/blob/master/CONTRIBUTING.md).

# Report

## What did you do?
I pod update and build.

## What did you expect to happen?
xcode build success.

## What happened instead?
xcode build failed.

### Stack

```
   CocoaPods : 1.4.0.rc.1
        Ruby : ruby 2.4.1p111 (2017-03-22 revision 58053) [x86_64-darwin17]
    RubyGems : 2.6.14
        Host : Mac OS X 10.13.2 (17C205)
       Xcode : 9.2 (9C40b)
         Git : git version 2.14.3 (Apple Git-98)
Ruby lib dir : /Users/admin/.rvm/rubies/ruby-2.4.1/lib
Repositories : master - https://github.com/CocoaPods/Specs.git @ 6b109f6d5c3ca99638bdc82062848e081d03d191
```

### Installation Source

```
Executable Path: /Users/admin/.rvm/gems/ruby-2.4.1/bin/pod
```

### Plugins

```
cocoapods-deintegrate : 1.0.1
cocoapods-plugins     : 1.0.0
cocoapods-search      : 1.0.0
cocoapods-stats       : 1.0.0
cocoapods-trunk       : 1.3.0
cocoapods-try         : 1.1.0
```

### Podfile

```ruby
# Uncomment the next line to define a global platform for your project
# platform :ios, '9.0'

target 'AAAA' do
  # Uncomment the next line if you're using Swift or would like to use dynamic frameworks
  use_frameworks!

  # Pods for AAAA
  pod 'A',:path => 'A.podspec'
  pod 'B',:path => 'B.podspec'

end
```

## Project that demonstrates the issue

[AAAA.zip](https://github.com/CocoaPods/CocoaPods/files/1648770/AAAA.zip)

**Comment by paulb777** [comment]
A dynamic framework cannot depend upon a static framework. 

If `s.static_framework      = true` is added to B.podspec, the build will succeed.

CocoaPods should generate an error message at `pod update/install` for this example.

**Comment by dnkoutso** [comment]
@paulb777 if you have time, we can fix this for 1.5.0
